AstarteDeviceSDKCSharp/Data/AstarteFailedMessageStorage.cs

```diff
@@ -172,5 +172,10 @@ public void AckFirstCache()
                 _astarteFailedMessageVolatile.Remove(failedMessages.First());
             }
         }
+
+        public bool IsExpired(long expire)
+        {
+            return expire > (DateTimeOffset.UtcNow.ToUnixTimeMilliseconds());
```

Comment:
This function is inverted, this expire until > of `DateTimeOffset.UtcNow.ToUnixTimeSeconds`, so or call this function isNotExpired, or invert the field:
```suggestion
            return DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() > expire;
```